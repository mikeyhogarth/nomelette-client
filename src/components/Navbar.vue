<template>
  <div id="nav">
    <img alt="Vue logo" src="../assets/nomelette-logo.png" />
    <div id="nav-links">
      <router-link to="/">Home</router-link>&nbsp;|
      <router-link to="/about">About</router-link>&nbsp;|
      <router-link to="/books">Books</router-link>&nbsp;|
      <router-link to="/browse">Browse</router-link>
    </div>
  </div>
</template>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="less">
#nav {
  padding: 30px;
  background-color: #00d2d1;
  color: #fff;
  a {
    font-weight: bold;
    color: #fff;
    &.router-link-exact-active {
      color: #ff5722;
    }
  }
}
</style>
