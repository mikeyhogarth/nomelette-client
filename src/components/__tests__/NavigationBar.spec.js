import { shallowMount, createLocalVue } from "@vue/test-utils";
import Navbar from "@/components/NavigationBar.vue";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("Navbar", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(Navbar, { localVue });
    expect(wrapper.element).toMatchSnapshot();
  });
});
