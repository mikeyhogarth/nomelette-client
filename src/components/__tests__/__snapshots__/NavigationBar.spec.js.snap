// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div
  id="nav"
>
  <img
    alt="Vue logo"
    src="../assets/images/nomelette-logo.png"
  />
   
  <div
    id="nav-links"
  >
    <router-link-stub
      event="click"
      tag="a"
      to="/"
    >
      Home
    </router-link-stub>
     |
    
    <router-link-stub
      event="click"
      tag="a"
      to="/about"
    >
      About
    </router-link-stub>
     |
    
    <router-link-stub
      event="click"
      tag="a"
      to="/books"
    >
      Books
    </router-link-stub>
     |
    
    <router-link-stub
      event="click"
      tag="a"
      to="/browse"
    >
      Browse
    </router-link-stub>
  </div>
</div>
`;
